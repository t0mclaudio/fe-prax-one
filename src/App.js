import { BrowserRouter } from "react-router-dom";
import AppContextProvider from "./context";
import Page from "./Page";
import AppRoutes from "./routes";

function App() {
  return (
    <AppContextProvider>
      <BrowserRouter basename={process.env.PUBLIC_URL}>
        <Page>
          <AppRoutes />
        </Page>
      </BrowserRouter>
    </AppContextProvider>
  );
}

export default App;
