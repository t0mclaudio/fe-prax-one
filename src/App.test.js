import { screen, render } from "@testing-library/react";
import App from "./App";

import renderer from 'react-test-renderer'

it('should create snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders page header and footer', () => {
  render(<App />);
  const header = screen.getByText('Home');
  expect(header).toBeInTheDocument();
})

it('renders page footer', () => {
  render(<App />);
  const footer = screen.getByText(/2021 TODO/i);
  expect(footer).toBeInTheDocument();
})