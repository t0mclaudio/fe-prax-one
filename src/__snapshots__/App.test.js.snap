// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create snapshot 1`] = `
<div
  className="page"
>
  <div
    className="header"
  >
    <header>
      <nav>
        <ul>
          <li>
            TODO APP
          </li>
          <li>
            <a
              aria-current="page"
              className="active"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              className="active"
              href="/"
              onClick={[Function]}
            >
              About
            </a>
          </li>
        </ul>
      </nav>
    </header>
  </div>
  <main
    className="main"
  >
    <section
      className="section"
    >
      <p
        className="nothingToShow"
      >
        Nothing to show
      </p>
    </section>
  </main>
  <div
    className="footer"
  >
    <footer>
      <p>
        Â© 2021 TODO
      </p>
    </footer>
  </div>
</div>
`;
